lib/modules/userInfo.js

```diff
@@ -101,61 +96,160 @@ module.options = {
 export const highlightedUsers = {};
 
 module.go = () => {
-	if (module.options.hoverInfo.value) {
-		$(document.body).on('mouseover', UserTagger.usernameSelector, handleMouseOver);
-	}
+	observe(document.body, { childList: true }, mutation => {
+		if (!mutation.addedNodes.length) return;
+		// author-tooltip
+		const addedNode = downcast(mutation.addedNodes[0], HTMLElement);
+		if (addedNode.classList.contains('author-tooltip')) {
+			setTimeout(() => renderHoverCardInfo(addedNode), 50);
+		}
+	});
 };
 
-function handleMouseOver(e: Event) {
-	const authorLink = downcast(e.target, HTMLAnchorElement);
-	if (!UserTagger.usernameRE.test(authorLink.href)) return;
-	const [, username] = UserTagger.usernameRE.exec(authorLink.href);
-	if (!username) console.error(i18n('userInfoInvalidUsernameLink'));
-	const thing = Thing.from(authorLink);
-
-	Hover.infocard(module.moduleID)
-		.target(e.target)
-		.options({
-			width: 475,
-			openDelay: module.options.hoverDelay.value,
-			fadeDelay: module.options.fadeDelay.value,
-			fadeSpeed: module.options.fadeSpeed.value,
-		})
-		.populateWith(card => showUserInfo(card, username, thing))
-		.begin();
-}
+function renderHoverCardInfo(hoverCard) {
+	const authorLink = hoverCard.querySelector('.author-tooltip__profile-link');
+	// Hopefully this will switch to event, but for now, just poll until its there
+	// this results in it being a ton less slow to render
+	if (!authorLink) {
+		return setTimeout(() => renderHoverCardInfo(hoverCard), 50);
+	}
 
-async function showUserInfo(card, username, thing) {
-	card.populate([username]);
+	const [, username] = UserTagger.usernameRE.exec(authorLink.getAttribute('href'));
+	applyResData(hoverCard, username);
+}
 
+/**
+ * applyResData
+ * Apply additional RES data to the reddit hover-cards
+ *
+ * @param Element hoverCard
+ * @param string username
+ */
+async function applyResData(hoverCard, username) {
 	let data;
 
 	try {
 		({ data } = await getUserInfo(username));
 	} catch (e) {
 		return [null, i18n('userInfoUserNotFound')];
 	}
-
 	if (data.is_suspended) {
 		return [null, i18n('userInfoUserSuspended')];
 	}
 
-	const d = new Date(data.created_utc * 1000);
+	// Get main elements
+	const header = hoverCard.querySelector('.author-tooltip__head');
+	const links = hoverCard.querySelector('.author-tooltip__link-list');
+
+	// Render tagger
+	const userTag = await attachUserTagger(header, username);
+	// Other
+	// attachGoldLogo(header, data);
+	// Extra buttons
+	createFriendButton(links, data);
+	createIgnoreButton(links, userTag);
+	createHighlightButton(links, username);
+}
+
+/**
+ * attachUserTagger
+ *
+ * @param Element header
+ * @param string username
+ */
+async function attachUserTagger(header: HTMLElement, username) {
+	const userTaggerEnabled = Modules.isRunning(UserTagger);
+	const userTag: Tag = userTaggerEnabled && await UserTagger.Tag.get(username);
+
+	if (userTaggerEnabled) {
```

Comment:
In general Flow doesn't do refinement "backwards", i.e. inference proceeds as:
- `userTaggerEnabled` inferred as `boolean` by its declaration
- `userTag` inferred as `false | <return value of Tag.get>` by its declaration
- `userTaggerEnabled` refined to `true` within the if-statement, but this doesn't propagate "backwards" to the existing declaration of `userTag`

You can either replace `if (userTaggerEnabled)` with `if (userTag)`, or change this to:

```js
if (Modules.isRunning(UserTagger)) {
    const userTag = await UserTagger.Tag.get(username);
    // ...
    return userTag;
}
return false;
```

which IMO is simpler for humans (not just Flow) to understand.

Reply:
Hmm, wonder how/why it worked initially as that was pretty much a copy and paste, prefer your suggestion tho, will update